### PR TITLE
allow child to be undefined or null

### PR DESCRIPTION
**Pull request by FredericHeem**
This PR solves #16, allowing simpler client's code.

### CODE REVIEW

**Review by Tao-VanJS — COMMENTED** [review]
I think there is one issue with the implementation, if the value of `child` is 0 (number), the will be filtered out.

I'm thinking of the implementation like:
```js
children.map(c => c ?? []).flat(Infinity).forEach(...)
```

Thanks so much for your contribution! I will do the merging this or next Friday, depending on my availability (currently only on Fridays I have convenient available personal time to work on this project).

**Comment by FredericHeem** [comment]
Good catch, the filtering should be placed after the flattening, shoudn't it ?

**Comment by Tao-VanJS** [comment]
After `null` or `undefined` is converted into `[]`, calling to `flat` will produce nothing.

For example
```js
[undefined, 1].map(c => c ?? []) // [[], 1]
```

```js
[[], 1].flat(Infinity) // [1]
```

**Comment by FredericHeem** [comment]
PR updated and ready for review.

File: src/van.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ let state = initVal => ({
 let toDom = v => v.nodeType ? v : new Text(v)
 
 let add = (dom, ...children) => (
-  children.flat(Infinity).map(c => c ?? []).forEach(child => dom.appendChild(
+  children.flat(Infinity).filter(c => c != undefined).forEach(child => dom.appendChild(
```
Comment (shuckster) [inline src/van.js:45]: `!= null` is shorter and achieves the same thing.

**Comment by Tao-VanJS** [comment]
This PR was manually merged in [0.12.1 release](https://github.com/vanjs-org/van/releases/tag/0.12.1). However, @FredericHeem doesn't show up as the contributor of this repo despite I explicitly listed @FredericHeem as the co-author of the [commit](https://github.com/vanjs-org/van/commit/f50ad3e33fabf3b90af757e14a5a493f1b91a4ba). Not sure why.

This method worked previously in [another commit](https://github.com/vanjs-org/van/commit/377175ce1dd906f12fb5a2517afbd55fb3265584), though.

Nonetheless, I manually added @FredericHeem in the [contributor list](https://github.com/vanjs-org/van#contributors) of the README.md file.

**Pull request closed without merging**